portfolio/src/main/webapp/script.js

```diff
@@ -48,9 +48,38 @@ addRandomCat = () => {
   else catContainer.appendChild(catimg);
 }
 
-window.onload = () => {
+/*
+ * Use Potion API to create page content for blog post 1 from a Notion document.
+ */
+getBlogPost = () => {
   fetch("https://potion-api.now.sh/html?id=" + POST_ID)
     .then(res => res.text())
     .then(text => { document.getElementById("post1").innerHTML = text; 
   });
 }
+
+/** 
+ * Fetch comments from server and insert them on blog page.
+ */
+getBlogComments = () => {
+  fetch("/data").then(response => response.json()).then((commentParts) => {
+    const commentsContainer = document.getElementById("submitted-comments-container");
+    commentsContainer.innerHTML = '';
+    commentParts.forEach(element => commentsContainer.appendChild(createParagraphElement(element)));
+  });
+}
+
+/*
+ * Fetch page data and set up HTML elements on load.
+ */
+window.onload = () => {
+  getBlogPost();
+  getBlogComments();
+}
+
+// Creates a <p> element containing text.
+createParagraphElement = (text) => {
+  const pElement = document.createElement('p');
+  pElement.innerText = text;
+  return pElement;
```

Comment:
also as an fyi, not sure how needed this is right now seems overkill, but one way to structure the js files once it grows (assuming no framework) is to have this in a utility file specifically for DOM.

ie in a `dom_util` you have this createParagraphElement (since its generic enough to be re-used across various pages (blogs, photos, homepage), and later on with unit-testing this will also be a unit-tested as a utility method instead of mixed-in with the UI business logic of fetching data/displaying blog, etc.

that can also get you started on having multiple js files and exporting/importing fn's, etc.

Reply:
that makes sense, thanks for the advice!